src/components/exchange/NetworkSwitcherv2.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useRef } from 'react';
+import { ScrollView } from 'react-native-gesture-handler';
+import RadialGradient from 'react-native-radial-gradient';
+import Divider from '../Divider';
+import { ButtonPressAnimation } from '../animations';
+import { CoinIcon } from '../coin-icon';
+import ChainBadge from '../coin-icon/ChainBadge';
+import { Box, Columns, Inline, Text } from '@/design-system';
+import { AssetType } from '@/entities';
+import { Network } from '@/helpers';
+import networkInfo from '@rainbow-me/helpers/networkInfo';
+import { ETH_ADDRESS, ETH_SYMBOL } from '@rainbow-me/references';
+import { position } from '@rainbow-me/styles';
+import { ethereumUtils } from '@rainbow-me/utils';
+import { useTheme } from '@/theme';
+
+const networkMenuItems = Object.values(networkInfo)
+  .filter(({ disabled, testnet }) => !disabled && !testnet)
+  .map(netInfo => ({
+    chainId: ethereumUtils.getChainIdFromNetwork(netInfo.value),
+    network: netInfo.value,
+    title: netInfo.name,
+    type: netInfo.value !== Network.mainnet ? netInfo.value : AssetType.token,
+  }));
+
+const NetworkSwitcherv2 = ({
+  hideDivider,
+  currentChainId,
+  setCurrentChainId,
+  testID,
+}: {
+  hideDivider: boolean;
+  currentChainId: number;
+  setCurrentChainId(chainId: number): void;
+  testID: string;
+}) => {
+  const { colors } = useTheme();
+  const scrollViewRef = useRef<ScrollView>(null);
+  const radialGradientProps = (network: Network) => {
+    return {
+      center: [0, 1],
+      colors: [
+        colors.alpha(colors.networkColors[network], 0.1),
+        colors.alpha(colors.networkColors[network], 0.02),
+      ],
+      pointerEvents: 'none',
+      style: {
+        ...position.coverAsObject,
+        overflow: 'hidden',
+      },
+    };
+  };
+
+  // if polygon is selcted intially we should scroll into view
+  useEffect(() => {
+    if (currentChainId === 137 && scrollViewRef?.current) {
+      console.log('triggering');
+      scrollViewRef?.current.scrollToEnd();
```

Comment:
nice touch! wondering if we should add additional behavior like this, for example on my android starting on optimism will leave the op Box hidden since the screen is small

Reply:
this makes sense to me, added